ext/pdo/pdo_dbh.c

```diff
@@ -1158,21 +1158,21 @@ PHP_METHOD(PDO, query)
 	pdo_dbh_object_t *dbh_obj = Z_PDO_OBJECT_P(ZEND_THIS);
 	pdo_dbh_t *dbh = dbh_obj->inner;
 
-	if (FAILURE == zend_parse_parameters(ZEND_NUM_ARGS(), "S|l!*", &statement,
-			&fetch_mode, &fetch_mode_is_null, &args, &num_args)) {
+	if (UNEXPECTED(FAILURE == zend_parse_parameters(ZEND_NUM_ARGS(), "S|l!*", &statement,
```

Comment:
using ZPP here _might_ make more a difference here.

Reply:
This method is used frequently, so it may be reasonable.